src/blockserver/server.py

```diff
@@ -105,7 +106,7 @@ def initialize(self, transfer_cls, get_auth_cls, get_cache_cls, database_pool,
         self.streamer = None
         await self._authorize_request()
         if self.request.method == 'POST':
-            self.request.connection.set_max_body_size(options.max_body_size)
+            self.remaining_upload_size = options.max_body_size
```

Comment:
I am not sure why max_body_size doesn't really work (on_finish is never called), my guess is that it works on the connection layer and aborts on a lower layer than the request processor, hence the request is not cleaned up.